opc-ua-sdk/sdk-server/src/main/java/org/eclipse/milo/opcua/sdk/server/services/helpers/MdnsHelper.java

```diff
@@ -0,0 +1,284 @@
+
+        Optional<InetAddress> highestPrioAddress = getMostPublicAddress(bindAddresses);
+
+        if (highestPrioAddress.isPresent()) {
+            ownAddress = highestPrioAddress.get();
+        } else {
+            try {
+                logger.warn("mDNS: no bind address valid. Setting to 0.0.0.0");
+                ownAddress = InetAddress.getByName("0.0.0.0"); // Create a JmDNS instance
+            } catch (UnknownHostException e) {
+                e.printStackTrace();
+            }
+        }
+
+        try {
+            jmdns = JmDNS.create(ownAddress);
+        } catch (IOException e) {
+            logger.error("Could not create instance of JmDNS: " + e.getMessage());
+            e.printStackTrace();
+        }
+        serverOnNetworkMap = new HashMap<>();
+        serverOnNetworkList = new LinkedList<>();
+        lastServerOnNetworkId = 0;
+        lastServerOnNetworkIdReset = new Date();
+    }
+
+    public void setMulticastServerConsumer(Consumer<ServerOnNetwork> multicastServerConsumer) {
+        this.multicastServerConsumer = multicastServerConsumer;
+    }
+
+
+    public boolean addRecord(String mdnsName, int port, String endpointName, String[] capabilities) {
+        if (endpointName.length() > 2 && endpointName.startsWith("/")) {
+            endpointName = endpointName.substring(1);
+        }
+
+        String txt = "path=/" + endpointName;
+        if (capabilities != null && capabilities.length > 0) {
+            txt += " caps=" + String.join(",", (CharSequence[]) capabilities);
+        }
+
+        ServiceInfo serviceInfo = ServiceInfo.create(mdnsListening, mdnsName, port, txt);
+        try {
+            jmdns.registerService(serviceInfo);
+        } catch (IOException e) {
+            logger.error("Could not add mDNS record for: " + mdnsName + " on port " + port + " with endpoint path: " +
+                endpointName + ". Error: " + e.getMessage());
+            return false;
+        }
+        return true;
+    }
+
+
+    public boolean removeRecord(String mdnsName, int port, String endpointName) {
+        if (endpointName.length() > 2 && endpointName.startsWith("/")) {
+            endpointName = endpointName.substring(1);
+        }
+
+        ServiceInfo serviceInfo = ServiceInfo.create(mdnsListening, mdnsName, port, "path=/" + endpointName);
+        jmdns.unregisterService(serviceInfo);
+        return true;
+    }
+
+    private String serviceInfoToDiscoveryUrl(ServiceInfo serviceInfo) {
+        String discoveryUrl = null;
+
+        InetAddress[] addressList = serviceInfo.getInetAddresses();
+        for (InetAddress mdnsIa : addressList) {
+
+            discoveryUrl = "opc.tcp://" + mdnsIa.getHostAddress() + ":" + serviceInfo.getPort() +
+                serviceInfo.getPropertyString("path");
+
+            if (discoveryUrl.endsWith("/")) {
+                discoveryUrl = discoveryUrl.substring(0, discoveryUrl.length() - 1);
+            }
+
+            break;
```

Comment:
Okay, so without the break, it just ends up setting `discoveryUrl` using the last element in `addressList`